java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -137,10 +136,19 @@ protected ArrowBuf releaseBuffer(ArrowBuf buffer) {
     return buffer;
   }
 
+  @Override
   public int getValueCount() { return 0; }
 
+  @Override
   public void setValueCount(int valueCount) { }
 
+  @Override
   public Object getObject(int index) { return null; }
+
+  @Override
+  public int getNullCount() { return 0; }
```

Comment:
Hmm.. I am not sure about these set of functions. What are the purpose of these?

Reply:
See my comment here that indicates why we have dummy getAccessor() and getMutator() interfaces https://github.com/apache/arrow/pull/1203/files/fb5768fd086a979af8cc8d8795935d191f231678#diff-b5610c173675c3c2707593d7968ee29eR259

In order to comply with top level ValueVector interface, I have just put these functions here. Once we remove other vectors (non-nullable) from ValueVector hierarchy, we can remove these methods too.

Reply:
These methods are not used anywhere.

Reply:
How about have them throw `UnsupportedOperationException`?

Reply:
I am concerned these methods would be confusing to user, so we have to keep them, I think we should make it clear these function won't work and shouldn't be called at all.

Reply:
This is resolved.